allanrogerr/operator
# Expose tenant certificates

## How to test

1. Clone https://github.com/allanrogerr/operator/tree/expose-tenant-certificates
2. Set testing on `./testing/deploy-tenant.sh`. e.g. `export OPERATOR_ENABLE_MANUAL_TESTING=on`
3. Deploy a tenant e.g. ../testing/deploy-tenant.sh
4. Using the console, add certificates under Security > Custom Certificates. See example of how to create valid and expired certificates here: https://github.com/allanrogerr/public/wiki/Console:-add-time-left-to-certificates-%231130
5. View tenant description using e.g. `k -n tenant-lite describe tenant/storage-lite`
6. Observe non-repetitive certificate issues (if they exist) exposed as events as well as in the status jsonpath

![image](https://user-images.githubusercontent.com/16472240/217102854-19db4414-71ed-4544-97b3-c00205c557b8.png)


<img width="1166" alt="image" src="https://user-images.githubusercontent.com/16472240/217102748-9640f0c1-1cfa-417f-a8ff-af17feab277b.png">

7. View events e.g. k get events -n tenant-lite
8. Observe non-repetitive certificate issues (if they exist) exposed as events

![image](https://user-images.githubusercontent.com/16472240/217102933-5c147f5c-49c8-401f-9915-143a71ad630a.png)
